airflow/cli/commands/celery_command.py

```diff
@@ -170,7 +170,7 @@ def worker(args):
         "--loglevel",
         celery_log_level,
         "--pidfile",
-        worker_pid_file_path,
+        worker_pid_file_path + ".celery.pid",
```

Comment:
If not mistaken this code is only for backward comptabitiy. Fixes should not be applied here but in:
https://github.com/apache/airflow/blob/c2fee42db05c2069c399e0d25f564110c920fea8/airflow/providers/celery/executors/celery_executor.py#L184

Reply:
@eladkal  I think provider package only loads command and valid args,
Usage of those args is actually in changed file only

Reply:
cc: @o-nikolas - indeed seems we are calling the commands from airflow - did we just miss it ? I can't see any reason why we could not move `celery_command.py` to provider ? But even if we could, I think that ship has sailed now, because Celery providers released so far expect it in airflow?

Reply:
Yupp, back when AIP-51 was being worked on initially, the celery executor loaded its commands from that `celery_command.py` module. That module should have been moved to the new provider package when that migration was underway afterwords. But looks like it just got missed in coding and in the PR! The same is likely true for the kubernetes executor.

Reply:
How should we procceed? Shall we create a copy of it in providers?

Reply:
Yeah, we could maintain two copies for now (or even automate this with a pre-commit) and eventually deprecate the core Airflow copy?

Reply:
@pavansharma36 ? Are you ok with those ideas? Is everything clear?

Reply:
@potiuk  so I just make copy in celery provider and make changes there?
Also PR is quick fix for problem should I just add one more arg to take something like --celery-pid ?

Reply:
Fix should be done in both places (Ideally, possibly we could make one of those - the provider one - the source of truth and copy the changes to the main from there (with pre-commit) and yes adding a parameter might be a good next step.

Reply:
Updated PR, since --pid arg in celery option was creating file with same pid as worker.pid, removed that arg to celery and generate pid only with run_command_with_daemon_option.

Updated executor to use celery command from provider package and deprecated celery_command in core airflow